traitsui/editors/shell_editor.py

```diff
@@ -50,16 +50,21 @@ def init(self, parent):
         value = self.value
         if self.factory.share and isinstance(value, dict):
             locals = value
-        self._shell = shell = PythonShell(parent, locals=locals)
+        self._shell = shell = PythonShell(parent)
         self.control = shell.control
+        shell.interpreter().locals = locals
```

Comment:
I don't think we should be overriding the `locals` dictionary completely. Given that the `IPythonShell` interface provides the `bind` method, I think we should use that to update the `locals` dictionary on the `interpreter` rather than setting it.

I am unable to follow the objects correctly but if `PythonWidget` is what gets used as the `control` for this editor, then it's `interpreter` gets initialized with the `__name__` and `__doc__` keys in the `locals` dictionary - which I think we might be discarding if we reset the dictionary.

Reply:
never mind. `__name__` and `__doc__` do exist in the locals namespace of the shell editor so this approach seems to work as expected.